util/xarray_overrides.py

```diff
@@ -159,6 +162,265 @@ def drop(self, labels, dim=None, inplace=False):
+            already loaded.
+          * 'all': All data variables will be concatenated.
+          * list of str: The listed data variables will be concatenated, in
+            addition to the 'minimal' data variables.
+        If objects are DataArrays, data_vars must be 'all'.
+    coords : {'minimal', 'different', 'all' o list of str}, optional
+        These coordinate variables will be concatenated together:
+          * 'minimal': Only coordinates in which the dimension already appears
+            are included.
+          * 'different': Coordinates which are not equal (ignoring attributes)
+            across all datasets are also concatenated (as well as all for which
+            dimension already appears). Beware: this option may load the data
+            payload of coordinate variables into memory if they are not already
+            loaded.
+          * 'all': All coordinate variables will be concatenated, except
+            those corresponding to other dimensions.
+          * list of str: The listed coordinate variables will be concatenated,
+            in addition the 'minimal' coordinates.
+    compat : {'equals', 'identical'}, optional
+        String indicating how to compare non-concatenated variables and
+        dataset global attributes for potential conflicts. 'equals' means
+        that all variable values and dimensions must be the same;
+        'identical' means that variable attributes and global attributes
+        must also be equal.
+    positions : None or list of integer arrays, optional
+        List of integer arrays which specifies the integer positions to which
+        to assign each dataset along the concatenated dimension. If not
+        supplied, objects are concatenated in the provided order.
+    indexers, mode, concat_over : deprecated
+
+    Returns
+    -------
+    concatenated : type of objs
+
+    See also
+    --------
+    merge
+    auto_combine
+    """
+    # Below TODOs are copied from the xarray source code
+    # TODO: add join and ignore_index arguments copied from pandas.concat
+    # TODO: support concatenating scalar coordinates even if the concatenated
+    # dimension already exists
+
+    try:
+        first_obj, objs = utils.peek_at(objs)
+    except StopIteration:
+        raise ValueError('must supply at least one object to concatenate')
+
+    if dim is None:
+        warnings.warn('the `dim` argument to `concat` will be required '
+                      'in a future version of xarray; for now, setting it to '
+                      "the old default of 'concat_dim'",
+                      FutureWarning, stacklevel=2)
+        dim = 'concat_dims'
+
+    if indexers is not None:  # pragma: nocover
+        warnings.warn('indexers has been renamed to positions; the alias '
+                      'will be removed in a future version of xarray',
+                      FutureWarning, stacklevel=2)
+        positions = indexers
+
+    if mode is not None:
```

Comment:
This shouldn't happen, because we're calling this function from stream engine only. Suggest removing the antiquated parameters.

Reply:
Will do.